larq_compute_engine/mlir/transforms/legalize_tflite.cc

```diff
@@ -0,0 +1,93 @@
+#include "flatbuffers/flexbuffers.h"
+#include "larq_compute_engine/mlir/ir/lce_ops.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "tensorflow/compiler/mlir/lite/ir/tfl_ops.h"
+
+namespace mlir {
+namespace TFL {
+
+namespace {
+
+std::vector<uint8_t> buildCustomOption(TF::BsignOp op) { return {}; }
+
+std::vector<uint8_t> buildCustomOption(TF::Bconv2dOp op) {
+  flexbuffers::Builder fbb;
+  fbb.Map([&]() {
+    fbb.Int("channels_in", op.channels_in().getSExtValue());
+    fbb.Int("dilation_height_factor",
+            op.dilation_height_factor().getSExtValue());
+    fbb.Int("dilation_width_factor", op.dilation_width_factor().getSExtValue());
+    fbb.String("fused_activation_function",
+               std::string(op.fused_activation_function()));
+    fbb.Int("pad_values", op.pad_values().getSExtValue());
+    fbb.String("padding", std::string(op.padding()));
+    fbb.Int("stride_height", op.stride_height().getSExtValue());
+    fbb.Int("stride_width", op.stride_width().getSExtValue());
+  });
+  fbb.Finish();
+  return fbb.GetBuffer();
+}
+
+std::vector<uint8_t> buildCustomOption(TF::BMaxPool2dOp op) {
+  flexbuffers::Builder fbb;
+  fbb.Map([&]() {
+    fbb.String("padding", std::string(op.padding()));
+    fbb.Int("stride_width", op.stride_width().getSExtValue());
+    fbb.Int("stride_height", op.stride_height().getSExtValue());
+    fbb.Int("filter_width", op.filter_width().getSExtValue());
+    fbb.Int("filter_height", op.filter_height().getSExtValue());
+  });
+  fbb.Finish();
+  return fbb.GetBuffer();
+}
```

Comment:
At some point it would be good to have unittests that check that we properly can roundtrip those attributes, for now I added a simple filecheck tests to verify that the defaults serialie properly.

Reply:
Added a unittest in c642235b153cc5e6d1f65adc21f3e2ee3dafbd2f and verified that the filecheck test prints the correct serialized options.